Neos.ContentRepository.Core/Classes/Service/ContentStreamPruner.php

```diff
@@ -24,47 +41,59 @@ class ContentStreamPruner implements ContentRepositoryServiceInterface
     public function __construct(
         private readonly ContentRepository $contentRepository,
         private readonly EventStoreInterface $eventStore,
+        private readonly EventNormalizer $eventNormalizer
     ) {
     }
 
     /**
-     * Remove all content streams which are not needed anymore from the projections.
+     * Before publishing version 3 (#5301) dangling content streams were not removed during publishing, discard or rebase
      *
-     * NOTE: This still **keeps** the event stream as is; so it would be possible to re-construct the content stream
-     *       at a later point in time (though we currently do not provide any API for it).
+     * Removes all nodes, hierarchy relations and content stream entries which are not needed anymore from the projections.
      *
-     *       To remove the deleted Content Streams,
-     *       call {@see ContentStreamPruner::pruneRemovedFromEventStream()} afterwards.
+     * NOTE: This still **keeps** the event stream as is; so it would be possible to re-construct the content stream at a later point in time.
      *
-     * By default, only content streams in STATE_NO_LONGER_IN_USE and STATE_REBASE_ERROR will be removed.
-     * If you also call with $removeTemporary=true, will delete ALL content streams which are currently not assigned
-     * to a workspace (f.e. dangling ones in FORKED or CREATED.).
+     * To prune the removed content streams from the event store, call {@see ContentStreamPruner::pruneRemovedFromEventStream()} afterwards.
      *
-     * @param bool $removeTemporary if TRUE, will delete ALL content streams not bound to a workspace
-     * @return iterable<int,ContentStreamId> the identifiers of the removed content streams
+     * @deprecated with Neos 9 beta 15, only used to migrate from earlier versions
      */
-    public function prune(bool $removeTemporary = false): iterable
+    public function removeDangelingContentStreams(\Closure $outputFn): void
```

Comment:
todo, this could be part of some integration check rather as with the new publishing one will never run into this case

Reply:
it cannot be part of the projection integrity things as ... well ... that is only for projections and if we forgot to cleanup after publishing its rather a core thing. Instead `flow contentStream:status` will help out ;)